connection.js

```diff
@@ -682,282 +626,142 @@ function conn_log() {
     }
 }
 
-function beforeRankedUnrankedGamesSpecial(before, extra, argNameString, special) {
-    const isExtra = (extra !== "");
-    if (argNameString.includes("unranked")) {
-        return `${before}${extra}unranked games`; //ex: "for blitz unranked games"
-    } else if (argNameString.includes("ranked")) {
-        return `${before}${extra}ranked games`;   //ex: "for ranked games"
-    } else if (isExtra) {
-        return `${before}${extra}games`           //ex: "for correspondence games"
-    } else if (special !== "") {
-        return `${before}${special}games`         //ex: "from all games"
-    } else {
-        return "";
-    }
+function generate_r_u_strings_connection(rankedSetting, speedSetting) {
+    const r_u = rankedSetting ? "unranked" : "ranked";
+    return { r_u,
+             for_r_u_games: `for ${r_u} games`,
+             for_blc_r_u_games: `for ${speedSetting} ${r_u} games` };
 }
 
-function rankToString(r) {
-    const R = Math.floor(r);
-    if (R >= 30)  return (R-30+1) + 'd'; // R>=30 : 1 dan or stronger
-    else          return (30-R) + 'k';   // R<30  : 1 kyu or weaker
-}
-
-function bannedFamilyReject(argNameString, uid, notificationUid) {
-    const rankedUnranked = beforeRankedUnrankedGamesSpecial("from ", "", argNameString, "all ");
-    conn_log(`${uid} ${notificationUid} is banned ${rankedUnranked}`);
-    return { reject: true, msg: `You (${uid} ${notificationUid}) are banned ${rankedUnranked} on this bot by bot admin, you may try changing the ranked/unranked setting` };
+function checkUserIsInSet(user, currSet) {
+    if (currSet.size === 0) return false;
```

Comment:
There is no need for this check as the line below will always be false for an empty set.

Reply:
indeed, fixed !!!